MCprep_addon/optimize_scene.py

```diff
@@ -180,7 +180,10 @@ def is_pricipled(self, context: Context, mat_type: str, node: Node) -> None:
 			if not roughness_socket.is_linked and roughness_socket.default_value >= 0.2:
 				self.glossy = self.glossy - 1 if self.glossy > 2 else 2
 		elif mat_type == "glass":
-			transmission_socket = node.inputs["Transmission"]
+			if "Transmission" in node.inputs:  # pre 4.0 way
+				transmission_socket = node.inputs["Transmission"]
+			else:  # Blender 4.0+
+				transmission_socket = node.inputs["Transmission Weight"]
```

Comment:
Personally, I would do the following (based on what [PEP8](https://peps.python.org/pep-0008/#maximum-line-length) regarding splitting lines):
```py
transmission_socket = (node.inputs["Transmission"] 
						if "Transmission in node.inputs 
						else node.inputs["Transmission Weight"])
```

However, what we have here is fine as well

Reply:
Fair suggestion, I tend to use ternary when the whole thing can fit on one line, otherwise I feel it's just extra messy to read with parentheses. Will leave as-is for now, thanks for reviewing.